src/raftstore/store/store.rs

```diff
@@ -1536,104 +1536,106 @@ impl<T: Transport, C: PdClient> Store<T, C> {
     fn on_ready_split_region(
         &mut self,
         region_id: u64,
-        left: metapb::Region,
-        right: metapb::Region,
-        right_derive: bool,
+        derived: metapb::Region,
+        regions: Vec<metapb::Region>,
     ) {
-        let (origin_region, new_region) = if right_derive {
-            (right.clone(), left.clone())
-        } else {
-            (left.clone(), right.clone())
+        let (peer_stat, is_leader) = match self.region_peers.get_mut(&region_id) {
```

Comment:
don't we need to check right_derive now?

Reply:
No. It doesn't matter if it's right_derive or not as long as the derived region can be found.